### ISSUE TITLE
Monitor validation set error during training

### ISSUE THREAD

**Issue by futurulus** [futurulus/coop-nets#1]
There should be an option to hold out a subset of the training set as a validation set to monitor generalization error. (Currently only training loss is monitored.)
